pkg/ingest/reader_client.go

```diff
@@ -57,7 +57,7 @@ func NewGroupReaderClient(kafkaCfg KafkaConfig, partitionRing ring.PartitionRing
 		kgo.SessionTimeout(3*time.Minute),
 		kgo.RebalanceTimeout(5*time.Minute),
 		kgo.Balancers(NewCooperativeActiveStickyBalancer(partitionRing)),
-		kgo.ConsumeResetOffset(kgo.NewOffset().AtStart()),
+		kgo.ConsumeResetOffset(kgo.NoResetOffset().AtEnd()),
```

Comment:
This makes the Kafka consumer start at the end of the topic (i.e. when no offsets are present) but stop consuming if it ever tries to read an offset that's out of range (which would happen if it fell behind the _entire_ backlog). In this case an operator needs to step in and reset offsets to what they think makes sense at that point.

`NewOffset().AtStart()` makes it start at the beginning of the topic, which is not reasonable when there's already a day worth of data present at first startup.

Reply:
definitely would like someone from rhythm to review this line

Reply:
I have removed this change from the diff. @mdisibio's changes in #4721 seems to be enough to let generators catch up even when there's substantial backlog.